cmd/playground/window_model.go

```diff
@@ -0,0 +1,207 @@
+type model struct {
+	cursor      cursorPointer
+	state       state
+	sizeInput   textinput.Model
+	offsetInput textinput.Model
+}
+
+func initModel() *model {
+	windowV1 := job.WindowV1{Size: defaultSize, Offset: defaultSize, TruncateTo: defaultTruncate}
+	windowV2 := job.WindowV2{Size: defaultSize, Offset: defaultSize, TruncateTo: defaultTruncate}
+	sizeInput := textinput.New()
+	sizeInput.Placeholder = defaultSize
+	sizeInput.Focus()
+	offsetInput := textinput.New()
+	offsetInput.Placeholder = defaultSize
+	return &model{
+		cursor:      pointToSize,
+		state:       state{windowV1: windowV1, windowV2: windowV2, scheduledTime: time.Now()},
+		sizeInput:   sizeInput,
+		offsetInput: offsetInput,
+	}
+}
+
+// since we don not intend to keep any input after the process is killed we return nil
+func (*model) Init() tea.Cmd {
+	return nil
+}
+
+// this handles motion of the pointer when we hit arrow-up key
+func (m *model) handleUp() {
+	switch m.cursor {
+	case pointToOffset:
+		// here we are shifting from Offsetinput to Sizeinput so we have to stop updating the Offsetinput(hence Offset.Blur())
+		// and start updating Sizeinput(hence Sizeinput.Focus())
+		m.offsetInput.Blur()
+		m.sizeInput.Focus()
+		m.cursor = pointToSize
+	case pointToTruncate:
+		m.offsetInput.Focus()
+		m.cursor = pointToOffset
+	case pointToYear, pointToMonth, PointToDay, pointToHour, pointToMinute:
+		m.cursor = pointToTruncate
+	}
+}
+
+func (m *model) handleDown() {
+	switch m.cursor {
+	case pointToTruncate:
+		m.cursor = pointToHour
+	case pointToOffset:
+		m.offsetInput.Blur()
+		m.cursor = pointToTruncate
+	case pointToSize:
+		m.offsetInput.Focus()
+		m.sizeInput.Blur()
+		m.cursor = pointToOffset
+	}
+}
+
+// handles left and right arrow key movements
+func (m *model) handleRight() {
+	switch m.cursor {
+	case pointToMonth:
+		m.cursor = pointToYear
+	case PointToDay:
+		m.cursor = pointToMonth
+	case pointToMinute:
+		m.cursor = PointToDay
+	case pointToHour:
+		m.cursor = pointToMinute
+	}
+}
+
+func (m *model) handleLeft() {
+	switch m.cursor {
+	case pointToMinute:
+		m.cursor = pointToHour
+	case PointToDay:
+		m.cursor = pointToMinute
+	case pointToMonth:
+		m.cursor = PointToDay
+	case pointToYear:
+		m.cursor = pointToMonth
+	}
+}
+func (m *model) handleIncrease() {
+	switch m.cursor {
+	case pointToTruncate:
+		m.state.IncrementTruncate()
+	case pointToYear, pointToMonth, PointToDay, pointToHour, pointToMinute:
+		m.state.IncrementTime(m.cursor.getStringValue())
+	}
+}
+func (m *model) handleDecrease() {
+	switch m.cursor {
+	case pointToTruncate:
+		m.state.DecrementTruncate()
+	case pointToYear, pointToMonth, PointToDay, pointToHour, pointToMinute:
+		m.state.DecrementTime(m.cursor.getStringValue())
+	}
+}
+
+func (m *model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	// switch currMsg := msg.(type) {
+	// case tea.KeyMsg:
+	currMsg := reflect.TypeOf(msg)
+	if currMsg.String() == "tea.KeyMsg" {
```

Comment:
instead of this, use the following:
```go
if currMsg.String() != "tea.KeyMsg" {
    return m, nil
}

switch fmt.Sprintf("%s", msg) {
...
```

This is to express that we care only if the message is "tea.KeyMsg", which in this case will be put as the top level flow, rather than lower level (or nested level).